### PR TITLE
fix qrCode state, add redirectUrl, add url

### PR DESCRIPTION
**Pull request by ApereLieZ**

### CODE REVIEW

File: app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/handlers/ext_int_query_proof_handler/ExtIntQueryProofHandler.kt [hunk 1]
```diff
@@ -82,7 +87,7 @@ fun ExtIntQueryProofHandler(
                     message = context.getString(R.string.query_proof_success_subtitle),
                 )
             )
-            onSuccess.invoke()
+            onSuccess.invoke(redirectURL)
```
Comment (ardier16) [inline app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/handlers/ext_int_query_proof_handler/ExtIntQueryProofHandler.kt:90]: we use `destination`, `redirect_uri` and `redirectURL` for the same param
let's stick to `redirectURI`

File: app/src/main/java/com/rarilabs/rarime/modules/main/MainScreen.kt [hunk 2]
```diff
@@ -93,12 +99,12 @@ fun AppLoadingScreen() {
...
-//            delay(500)
-//            scale.floatValue = 1f
-//            delay(500)
-//        }
+        while (true) {
+            scale.floatValue = 1.1f
+            delay(500)
+            scale.floatValue = 1f
+            delay(500)
+        }
```
Comment (ardier16) [inline app/src/main/java/com/rarilabs/rarime/modules/main/MainScreen.kt:107]: what's the purpose of this?

Comment (ardier16) [reply app/src/main/java/com/rarilabs/rarime/modules/main/MainScreen.kt:107]: if it's intentional, let's leave an explanation comment

**Review by ardier16 — APPROVED** [review]

File: app/src/main/java/com/rarilabs/rarime/modules/main/MainScreen.kt (hunk 2 above)
Comment (ApereLieZ) [reply app/src/main/java/com/rarilabs/rarime/modules/main/MainScreen.kt:107]: This is for unboarding animation.

File: app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/handlers/ext_int_query_proof_handler/ExtIntQueryProofHandler.kt (hunk 1 above)
Comment (ApereLieZ) [reply app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/handlers/ext_int_query_proof_handler/ExtIntQueryProofHandler.kt:89]: done

File: app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/handlers/light_proof_handler/LightProofHandlerViewModel.kt [hunk 3]
```diff
@@ -209,7 +213,7 @@ class LightProofHandlerViewModel @Inject constructor(
         )
     }
 
-    suspend fun loadDetails(proofParamsUrl: String): Map<String, String> {
+    suspend fun loadDetails(proofParamsUrl: String, redirectionURL: String?): Map<String, String> {
```
Comment (ardier16) [inline app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/handlers/light_proof_handler/LightProofHandlerViewModel.kt:216]: redirectUri

File: app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/handlers/light_proof_handler/LightProofHandlerViewModel.kt [hunk 4]
```diff
@@ -272,6 +276,11 @@ class LightProofHandlerViewModel @Inject constructor(
             Log.e("nationality", e.message, e)
         }
 
+        if (redirectionURL != null) {
+            tempMap["Redirection URL"] = redirectionURL
```
Comment (ardier16) [inline app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/handlers/light_proof_handler/LightProofHandlerViewModel.kt:280]: redirectUri

File: app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/handlers/ext_int_query_proof_handler/ExtIntQueryProofHandlerViewModel.kt [hunk 5]
```diff
@@ -53,7 +53,7 @@ class ExtIntQueryProofHandlerViewModel @Inject constructor(
     val fieldsParams: StateFlow<Map<String, String>>
         get() = _fieldsParams.asStateFlow()
 
-    suspend fun loadDetails(proofParamsUrl: String) {
+    suspend fun loadDetails(proofParamsUrl: String, redirectUrl: String) {
```
Comment (ardier16) [inline app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/handlers/ext_int_query_proof_handler/ExtIntQueryProofHandlerViewModel.kt:56]: seems like redirectUrl is unused

File: app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/components/HandlerPreviewerLayout.kt [hunk 6]
```diff
@@ -168,10 +160,20 @@ private fun HandlerPreviewerLayoutContent(
 
             if (previewFields.isNotEmpty()) {
                 previewFields.forEach { (key, value) ->
-                    HandlerPreviewerLayoutRow(
-                        key = key,
-                        value = value
-                    )
+                    if (key == "Redirection URL") {
```
Comment (ardier16) [inline app/src/main/java/com/rarilabs/rarime/api/ext_integrator/ext_int_action_preview/components/HandlerPreviewerLayout.kt:163]: uri

File: app/src/main/java/com/rarilabs/rarime/util/StringUtil.kt [hunk 7]
```diff
@@ -30,4 +31,36 @@ object StringUtil {
...
+
+            // Extract the substring starting after "://"
+            val remainder = url.substring(startIndex + schemeDelimiter.length)
+
+            // Find the first "/" after the scheme to isolate the target
+            val endIndex = remainder.indexOf("/")
+            return if (endIndex == -1) remainder else remainder.substring(0, endIndex)
+        } catch (e: Exception) {
+            // Log errors for invalid input
+            println("Error extracting target: ${e.message}")
```
Comment (ardier16) [inline app/src/main/java/com/rarilabs/rarime/util/StringUtil.kt:62]: logger

**Review by ardier16 — APPROVED** [review]

**Pull request merged**